middleware/middleware.js

```diff
@@ -104,3 +104,36 @@ module.exports.getPixelatedImage = async (image, difficulty = 3) => {
     .getBase64Async(Jimp.MIME_JPEG);
   return pixelatedImage;
 };
+
+module.exports.calculateScore = (session, sessionEnd) => {
+  const difficultyModifier = 0.5 + 0.5 * session.difficulty;
+  const msPerGuess = 15000;
+  const baseGuessScore = 1000;
+  const baseTimeBonusPoints = 50;
+
+  let correctAnswers = 0;
+
+  session.sessionData.forEach((guess) => {
+    if (guess.userGuess === true) {
+      correctAnswers++;
+    }
+  });
+
+  const guessAccuracy = correctAnswers / session.length;
+  const guessesScore = baseGuessScore * difficultyModifier * correctAnswers;
+  const bonusTime =
+    msPerGuess * session.length - (sessionEnd - session.sessionStartTime);
+
+  if (bonusTime <= 0) {
+    return guessesScore;
+  }
+
+  const timeBonusScore = Math.floor(
+    (bonusTime / 1000) *
+      difficultyModifier *
+      baseTimeBonusPoints *
+      guessAccuracy
+  );
+
+  return guessesScore + timeBonusScore;
```

Comment:
I think from code readability side this could be improved a bit if you wrote it like this:

```
  if (bonusTime <= 0) {
    return guessesScore;
  } else {
    const timeBonusScore = Math.floor(
    (bonusTime / 1000) *
      difficultyModifier *
      baseTimeBonusPoints *
      guessAccuracy
  );

  return guessesScore + timeBonusScore;
  }
  ```